components/script/dom/bindings/str.rs

```diff
@@ -306,11 +306,19 @@ impl DOMString {
     /// A valid number is the same as what rust considers to be valid,
     /// except for +1., NaN, and Infinity.
     /// https://html.spec.whatwg.org/multipage/#valid-floating-point-number
-    pub fn is_valid_number_string(&self) -> bool {
-        let input = &self.0;
-        input.parse::<f64>().ok().map_or(false, |val| {
-            !(val.is_infinite() || val.is_nan() || input.ends_with(".") || input.starts_with("+"))
-        })
+    pub fn is_valid_float_number_string(&self) -> bool {
+        parse_float_number(&self.0).is_ok()
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#best-representation-of-the-number-as-a-floating-point-number
+    pub fn best_representation_of_the_float_number(&mut self) {
+        let input = &self.0.clone();
```

Comment:
I think you can do same way as `is_valid_float_number_string`.
Just passing `&self.0` into `parse_float_number` then no need to `clone` here.

Reply:
` is_valid_float_number_string` don't need to change `self`.
However, here we change it. Need clone it to avoid conflict.

Reply:
oh, maybe I'm not familiar with borrow checker enough.
If you passing `self.0` with reference, will the variable be moved?